### PR TITLE
[FEATURE]: Remove GRPC and Refactor

### PR DESCRIPTION
**Pull request by SeedyROM**
# Tasks:

- [x] Remove all references to GRPC.
- [x] Remove all the cosmos client traits and structs.
- [x] Refactor components to no longer mention GRPC.
- [x] Rinse and repeat...

### CODE REVIEW

**Comment by mikedotexe** [comment]
Looking good! Amazing!

I saw something I'll point out, tho I know this is a draft PR.

I ran this command on a recent deployment of the the CronCat manager on Juno testnet:

    junod q wasm contract-state smart juno1gqkv06dxrccavckw8ydwaxm353pvlrtx0cgxfehvn0gjvlwjfscq58nn8w '{"get_agent_ids":{}}' --node https://juno-testnet-rpc.polkachu.com:443 --chain-id uni-5 -o json | jq

and, as expected, it spit out the agents, showing that my agent (that ends in `red` because I'm hardcore Red team) wasn't there:

```json
{
  "data": {
    "active": [
      "juno1rez0cc8zx8u75wqaz04xzcr83f79lw4hk62z7t"
    ],
    "pending": [
      "juno1znqkcerkhd3f78mwmrw30czl3xq46qyx4cq9h0"
    ]
  }
}
```

Then, on this branch, I ran:

    cargo run register --chain-id uni-5

And it looks like it registered, and then tried to register again:

```
                                         _
           ___ _ __ ___  _ __   ___ __ _| |_
  /\_/\   / __| '__/ _ \| '_ \ / __/ _` | __/
 ( o.o ) | (__| | | (_) | | | | (_| (_| | |_
  > ^ <   \___|_|  \___/|_| |_|\___\__,_|\__\
==============================================

2023-01-19T21:34:44.452809Z  INFO croncat::rpc::service: [uni-5] Picking best source for chain...
2023-01-19T21:34:45.344446Z  INFO croncat::rpc::service: [uni-5] 2 source(s) available!
2023-01-19T21:34:45.508834Z  INFO croncatd: Account ID: juno1hntaw5cgvxz0gftpd9gec876n0l6az20clrred
2023-01-19T21:34:50.065830Z  INFO croncatd: Account ID: juno1hntaw5cgvxz0gftpd9gec876n0l6az20clrred
2023-01-19T21:34:50.887618Z ERROR croncatd: Command failed: register
2023-01-19T21:34:50.887651Z ERROR croncatd: Agent already registered
```

**Comment by mikedotexe** [comment]
[UPDATE: resolved]

After the previous command I tried:

    cargo run go --chain-id uni-5

and looks like the agent status is being weird, hmm…

```
                                         _
           ___ _ __ ___  _ __   ___ __ _| |_
  /\_/\   / __| '__/ _ \| '_ \ / __/ _` | __/
 ( o.o ) | (__| | | (_) | | | | (_| (_| | |_
  > ^ <   \___|_|  \___/|_| |_|\___\__,_|\__\
==============================================

2023-01-19T21:41:03.276934Z  INFO croncat::rpc::service: [uni-5] Picking best source for chain...
2023-01-19T21:41:04.200359Z  INFO croncat::rpc::service: [uni-5] 2 source(s) available!
2023-01-19T21:41:06.107520Z ERROR croncatd: Command failed: go
2023-01-19T21:41:06.107551Z ERROR croncatd: Failed to deserialize response data: unknown variant `status`, expected one of `Active`, `Pending`, `Nominated` at line 1 column 9
```

**Comment by mikedotexe** [comment]
> 2023-01-19T21:34:45.508834Z  INFO croncatd: Account ID: juno1hntaw5cgvxz0gftpd9gec876n0l6az20clrred
> 2023-01-19T21:34:50.065830Z  INFO croncatd: Account ID: juno1hntaw5cgvxz0gftpd9gec876n0l6az20clrred
> 2023-01-19T21:34:50.887618Z ERROR croncatd: Command failed: register
> 2023-01-19T21:34:50.887651Z ERROR croncatd: Agent already registered

I feel like it might be related to this `loop`
I'm throwing lots of debug logs
https://github.com/CronCats/croncat-rs/pull/107/files#diff-3fb33f274778a4cc6618dc8a1f756a283612698c8536ccafe408dc44cf50b423R142

Best way to trigger it is to try one of these commands (whichever one will succeed once)

```rs
cargo run register --chain-id uni-5
// and
cargo run unregister --chain-id uni-5
```

Seems like it'll call it until it gets an error

**Comment by mikedotexe** [comment]
In case it's helpful, I made a "filthy logs" branch you can check out and see a bunch of "aloha value of variable is…"

Compare:
https://github.com/CronCats/croncat-rs/compare/feature/grpc-removal-filthy-logs

Branch:
https://github.com/CronCats/croncat-rs/tree/feature/grpc-removal-filthy-logs

You can see that when you call register/unregister there are several places that might help indicate where it's calling the contract twice

File: croncat/src/config.rs [hunk 1]
```diff
@@ -74,7 +71,7 @@ struct RawChainConfigEntry {
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct ChainDataSource {
-    pub grpc: String,
+    pub grpc: Option<String>,
```
Comment (TrevorJTClarke) [inline croncat/src/config.rs]: Is this just so things dont error? prolly best to remove

File: croncat/src/rpc/service.rs [hunk 2]
```diff
@@ -192,8 +184,7 @@ impl GrpcClientService {
                     },
                 )),
                 GrpcCallType::Query => GrpcClient::Query(Box::new(
-                    match GrpcQuerier::new(self.chain_config.clone(), source.grpc.to_string()).await
-                    {
+                    match Querier::new(self.chain_config.clone(), source.rpc.to_string()).await {
                         Ok(client) => client,
                         Err(e) => {
                             debug!("Failed to create grpc client for {}: {}", source_key, e);
```
Comment (TrevorJTClarke) [inline croncat/src/rpc/service.rs]: tiny - but could remove the `g` here ;)

File: croncat/src/rpc/signer.rs [hunk 3]
```diff
@@ -17,150 +19,134 @@ use cw_rules_core::msg::QueryConstruct;
...
             .public_key()
-            .account_id(&client.chain_info.bech32_prefix)?;
+            .account_id(&cfg.info.bech32_prefix)?;
+
+        // Create a new RPC client
+        let mut rpc_client = RpcClient::new(&cfg, rpc_url.as_str())?;
+        // TODO: This is a hack to get around the fact that cosm-tome doesn't
+        // let us pass an xprv.
+        rpc_client.set_mnemonic(mnemonic);
+        rpc_client.set_denom(&cfg.info.fees.fee_tokens[0].denom);
```
Comment (TrevorJTClarke) [inline croncat/src/rpc/signer.rs]: in the future, we can also use a config variable right? 
Chain config has a higher level one as well above fee_tokens called `base`

File: croncat/src/rpc/signer.rs [hunk 4]
```diff
@@ -17,150 +19,134 @@ use cw_rules_core::msg::QueryConstruct;
...
     {
-        let out = timeout(
-            Duration::from_secs(30),
-            self.client
-                .query_client
-                .query_contract(&self.manager.to_string(), msg),
-        )
-        .await
-        .map_err(|err| eyre!("Timeout (30s) while querying contract: {}", err))??;
+        let out = timeout(Duration::from_secs(30), self.rpc_client.wasm_query(msg))
```
Comment (TrevorJTClarke) [inline croncat/src/rpc/signer.rs]: This likely needs a shorter configurable timeout. Chain blocks are 6 seconds, so 30seconds is too far behind

File: croncat/src/rpc/signer.rs [hunk 5]
```diff
@@ -17,150 +19,134 @@ use cw_rules_core::msg::QueryConstruct;
...
-            Duration::from_secs(30),
-            self.client.execute_wasm(msg, &self.manager.to_string()),
-        )
-        .await
-        .map_err(|err| eyre!("Timeout (30s) while executing wasm: {}", err))??;
+    pub async fn execute_croncat<S>(&self, msg: S) -> Result<ChainResponse, Report>
+    where
+        S: Serialize,
+    {
+        let res = timeout(Duration::from_secs(30), self.rpc_client.wasm_execute(msg))
```
Comment (TrevorJTClarke) [inline croncat/src/rpc/signer.rs]: same timeout issue here, less than block time needed

File: croncat/src/rpc/querier.rs [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        f.debug_struct("Querier")
+            .field("croncat_addr", &self.croncat_addr)
+            .finish()
+    }
+}
+
+impl Querier {
+    pub async fn new(cfg: ChainConfig, rpc_url: String) -> Result<Self, Report> {
+        let rpc_url = if !rpc_url.starts_with("https://") {
+            format!("https://{}", rpc_url)
```
Comment (TrevorJTClarke) [inline croncat/src/rpc/querier.rs]: what happens if `http://`?

File: croncat/src/rpc/querier.rs [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+            .await?;
+
+        if agent_info.is_none() {
+            Err(eyre!("Agent not registered"))
+        } else {
+            Ok(agent_info.unwrap().status)
+        }
+    }
+
+    pub async fn get_tasks(
```
Comment (TrevorJTClarke) [inline croncat/src/rpc/querier.rs:84]: This might be missing query for tasks with rules

**Review by TrevorJTClarke — APPROVED** [review]
Hey @SeedyROM!! Tons of stuff here, thank you for this overhaul!

I left a few non-blocker comments. I think ill merge

File: croncat/src/config.rs (hunk 1 above)
Comment (SeedyROM) [reply croncat/src/config.rs]: :+1:

File: croncat/src/rpc/service.rs (hunk 2 above)
Comment (SeedyROM) [reply croncat/src/rpc/service.rs]: :+1:

File: croncat/src/rpc/querier.rs (hunk 6 above)
Comment (SeedyROM) [reply croncat/src/rpc/querier.rs]: Good call on this.

File: croncat/src/rpc/signer.rs [hunk 8]
```diff
@@ -70,28 +68,23 @@ impl Signer {
...
     pub async fn query_croncat<R, S>(&self, msg: S) -> Result<R, Report>
     where
         S: Serialize,
         R: DeserializeOwned,
     {
-        let out = timeout(Duration::from_secs(30), self.rpc_client.wasm_query(msg))
-            .await
-            .map_err(|err| eyre!("Timeout (30s) while querying contract: {}", err))??;
+        let out = timeout(
+            Duration::from_secs_f64(self.rpc_client.timeout_secs),
```
Comment (TrevorJTClarke) [inline croncat/src/rpc/signer.rs:77]: this is GREAT! thank you!

File: croncat/src/rpc/signer.rs [hunk 9]
```diff
@@ -61,7 +55,11 @@ impl Signer {
         // TODO: This is a hack to get around the fact that cosm-tome doesn't
         // let us pass an xprv.
         rpc_client.set_mnemonic(mnemonic);
-        rpc_client.set_denom(&cfg.info.fees.fee_tokens[0].denom);
+        rpc_client.set_denom(
+            cfg.denom
```
Comment (TrevorJTClarke) [inline croncat/src/rpc/signer.rs:59]: this is also GREAT! thank you!

**Review by TrevorJTClarke — APPROVED** [review]
im happy

**Pull request merged**